src/rest/subhandlers/rest_meta_info.erl

```diff
@@ -46,43 +46,85 @@ init(Req0, State) ->
 allowed_methods(Req, State) ->
     {[<<"GET">>, <<"POST">>, <<"PUT">>], Req, State}.
 
-content_types_provided(Req, State) ->
-    {[{<<"application/json">>, to_json}], Req, State}.
-
 content_types_accepted(Req, State) ->
     {[{<<"application/json">>, from_json}], Req, State}.
 
-%% @todo implement case for dependent_zone
+content_types_provided(Req, State) ->
+    {[{<<"application/json">>, to_json}], Req, State}.
+
+%% @todo implement case for dependent_zones, if needed
 resource_exists(Req, State) ->
+    {true, Req, State}.
+
+%% POST/PUT handler
+from_json(Req, State) ->
     Result = case maps:get(op, State) of
-                 dependent_zone -> some_logic;
-                 _ -> true
+                 instrumentation ->
+                     {ok, Bool} = parse_boolean(maps:get(enabled, State)),
+                     set_instrumentation(Bool);
+                 indexation ->
+                     {ok, Bool} = parse_boolean(maps:get(enabled, State)),
+                     set_indexation(Bool);
```

Comment:
Powtarzasz się trochę, więc pewnie warto by było znaleźć tu jakąś wspólną funkcję, na przykład tak
```suggestion
                 instrumentation ->
                     handle_setting(State, fun set_instrumentation/1);
                 indexation ->
                     handle_setting(State, fun set_indexation/1);
```

Reply:
Owszem, być może by się przydało, natomiast to zależy od tego jak będzą wyglądać funkcje przyszłego modułu settings.erl

Reply:
Niby tak, ale to są tylko dwie analogiczne linijki. "Naprawa" tego zajmie więcej miejsca, zwłaszcza że nie ma tu perspektyw na dojście kolejnych analogicznie obsługiwanych ustawień.

Więc osobiście bym to po prostu tak zostawił

Reply:
Tu nie chodzi o miejsce, tylko o to, że powtarzasz fragment kodu, który robi dokładnie to samo i później wrzucasz to do kolejnej funkcji. No ewentualnie może zostać, ale to nie jest ani eleganckie, ani funkcyjne